### PR TITLE
Allow users to update the final answer prompt of MagenticOne orc.

### PR DESCRIPTION
**Pull request by afourney**
In some cases (e.g., GAIA) we need the final output to be summarized or output in a particular way. This PR allows users to overwrite the final_answer_prompt.

### CODE REVIEW

File: python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_group_chat.py [hunk 1]
```diff
@@ -64,6 +65,7 @@ def __init__(
         termination_condition: TerminationCondition | None = None,
         max_turns: int | None = 20,
         max_stalls: int = 3,
+        final_answer_prompt: str = ORCHESTRATOR_FINAL_ANSWER_PROMPT,
```
Comment (ekzhu) [inline python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_group_chat.py:68]: Need an update on the API doc

File: python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_group_chat.py [hunk 2]
```diff
@@ -77,6 +79,7 @@ def __init__(
             raise ValueError("At least one participant is required for MagenticOneGroupChat.")
         self._model_client = model_client
         self._max_stalls = max_stalls
+        self._final_answer_prompt = final_answer_prompt
```
Comment (ekzhu) [inline python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_group_chat.py:82]: We will need to cover public-facing API features in unit tests in a separate PR: #4466

**Review by husseinmozannar — APPROVED** [review]
looks good to me once API documentation is updated in magentic_one_group_chat and we can tests in future PR

File: python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_group_chat.py (hunk 1 above)
Comment (afourney) [reply python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_group_chat.py:69]: Added

**Comment by afourney** [comment]
> looks good to me once API documentation is updated in magentic_one_group_chat and we can tests in future PR

Done.

**Pull request merged**